x/fees/client/cli/tx.go

```diff
@@ -0,0 +1,160 @@
+	)
+	return txCmd
+}
+
+// NewRegisterDevFeeInfo returns a CLI command handler for registering a
+// contract for fee distribution
+func NewRegisterDevFeeInfo() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "register-fee [contract_hex] [withdraw_bech32]",
+		Short: "Register a contract for fee distribution",
+		Args:  cobra.RangeArgs(1, 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			var withdraw string
+			deployer := cliCtx.GetFromAddress()
+
+			contract := args[0]
+			if err := ethermint.ValidateAddress(contract); err != nil {
+				return fmt.Errorf("invalid contract hex address %w", err)
+			}
+
+			if len(args) == 2 {
+				withdraw = args[1]
+				if _, err := sdk.AccAddressFromBech32(withdraw); err != nil {
+					return fmt.Errorf("invalid withdraw bech32 address %w", err)
+				}
+			}
+
+			msg := &types.MsgRegisterDevFeeInfo{
+				ContractAddress: contract,
+				DeployerAddress: deployer.String(),
+				WithdrawAddress: withdraw,
+			}
+
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(cliCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+// NewCancelDevFeeInfo returns a CLI command handler for canceling a
+// contract for fee distribution
+func NewCancelDevFeeInfo() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "cancel-fee [contract_hex]",
+		Short: "Cancel a contract from fee distribution",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			deployer := cliCtx.GetFromAddress()
+
+			contract := args[0]
+			if err := ethermint.ValidateAddress(contract); err != nil {
+				return fmt.Errorf("invalid contract hex address %w", err)
+			}
+
+			msg := &types.MsgCancelDevFeeInfo{
+				ContractAddress: contract,
+				DeployerAddress: deployer.String(),
+			}
+
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(cliCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+// NewUpdateDevFeeInfo returns a CLI command handler for updating the withdraw
+// address of a contract for fee distribution
+func NewUpdateDevFeeInfo() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-fee [contract_hex] [withdraw_bech32]",
+		Short: "Update withdraw address for a contract registered for fee distribution",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			deployer := cliCtx.GetFromAddress()
+
+			contract := args[0]
+			if err := ethermint.ValidateAddress(contract); err != nil {
+				return fmt.Errorf("invalid contract hex address %w", err)
+			}
+
+			withdraw := args[1]
```

Comment:
Error if deployer = withdraw

Reply:
`msg.ValidateBasic()` is used before broadcasting the tx. And this already checks `withdraw ≠ deployer` https://github.com/tharsis/evmos/blob/fe29d7d6f6756fa076f126652e7ce1cdcbccb74c/x/fees/types/msg.go#L177-L179. It also does the address validation, so these checks in cli/tx.go seem to be redundant.